app/mixins/controller/billing.js

```diff
@@ -0,0 +1,35 @@
+import Mixin from '@ember/object/mixin';
+import { reads } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+
+export default Mixin.create({
+  storage: service(),
+  router: service(),
+
+  account: reads('model.account'),
+  newSubscription: reads('model.newSubscription'),
+  queryParams: ['billingStep'],
+  billingStep: 1,
+  lastStep: 3,
+
+  init() {
+    this.router.on('routeDidChange', () => {
+      if (this.billingStep !== this.storage.billingStep) {
+        this.storage.clearBillingData();
+      }
+    });
```

Comment:
I'm a little bit concerned with this listener. It adds the handler when a user visits subscription tab, but it never removes the handler and the handler itself doesn't check the route. Thus, it looks like it'll execute on each transition, even outside the subscription tab, until page reload. I don't think it's expected.